lib/schema.js

```diff
@@ -428,21 +464,41 @@ class Metaschema {
           value = def(value);
           obj[field] = value;
         } else {
-          value = this.createInstance(def, value);
+          if (def.category) value = this.createInstance(def.category, value);
+          if (def.domain) value = this.createDomainInstance(def.domain, value);
           if (value !== null) obj[field] = value;
           else return null;
         }
       }
-      for (const prop of required) {
-        if (obj[prop] === null || obj[prop] === undefined) {
-          return null;
+      for (const prop of properties) {
+        const def = definition[prop];
+        if (!obj.hasOwnProperty(prop)) {
```

Comment:
If `obj = { A: undefined }` but `A` is `required` this won't catch that and just ignore this property, is that ok?

Reply:
IMHO, it is not, we should prohibit `null` or `undefined`.